### DISCUSSION TITLE
How can i catch server error ?

### DISCUSSION THREAD

**Discussion by MuttakinHasib** [rpldy/react-uploady#390] [Q&A]
1) how can I get the upload progress to the server?
2) how can I catch the server errors?

@yoavniran Please help me

**Comment by yoavniran** [comment]
Hi @MuttakinHasib 
you didnt provide details so its a little difficult to answer accurately.

When working with React, the best way to get progress and errors is using the Item's event hooks (ie: [useItemProgressListener](https://react-uploady.org/docs/api/hooks/useItemProgressListener/) & [useItemErrorListener](https://react-uploady.org/docs/api/hooks/useItemErrorListener/).

You can see a simple example of progress usage in this [sandbox](https://codesandbox.io/s/preview-progress-react-uploady-pki4i).

And you can see both in action in this more advanced [sandbox](https://codesandbox.io/s/react-uploady-with-antd-upload-with-preview-progress-d7o3w).

Let me know if there's something more specific I can help with.
